modules/webbrowser/src/processors/webbrowserprocessor.cpp

```diff
@@ -86,27 +89,52 @@ WebBrowserProcessor::WebBrowserProcessor()
 
     addProperty(sourceType_);
     addProperty(fileName_);
+    addProperty(autoReloadFile_);
     addProperty(url_);
     url_.setVisible(false);
     addProperty(reload_);
 
-    sourceType_.onChange([&]() {
-        switch (sourceType_.get()) {
-            default:
-            case SourceType::LocalFile:
-                fileName_.setVisible(true);
-                url_.setVisible(false);
-                break;
-            case SourceType::WebAddress:
-                fileName_.setVisible(false);
-                url_.setVisible(true);
-                break;
+    addProperty(runJS_);
+    addProperty(js_);
+
+    runJS_.onChange([this]() {
+        browser_->GetMainFrame()->ExecuteJavaScript(js_.get(), "", 1);
```

Comment:
Script should probably be executed in the process function to make sure that the web page is finished loading when pressing the button.